src/main/java/org/broadinstitute/consent/http/resources/TDRResource.java

```diff
@@ -64,4 +86,64 @@ public Response getDatasetByIdentifier(@Auth AuthUser authUser, @PathParam("iden
             return createExceptionResponse(e);
         }
     }
+
+    @POST
+    @Consumes("application/json")
+    @Produces("application/json")
+    @Path("/dar/draft")
+    @PermitAll
+    public Response createDraftDataAccessRequest(
+        @Auth AuthUser authUser,
+        @Context UriInfo info,
+        @QueryParam("identifiers") String identifiers,
+        @QueryParam("projectTitle") String projectTitle) {
+      try {
+        // Ensure that the user is a registered DUOS and SAM user
+        User user = userService.findOrCreateUser(authUser);
+        if (Objects.isNull(identifiers) || identifiers.isBlank()) {
+          throw new BadRequestException("No dataset identifiers were provided");
+        } else {
+          List<Integer> identifierList = Arrays.stream(identifiers.split(","))
+                  .map(String::trim)
+                  .filter(identifier -> !identifier.isBlank())
+                  .map(Dataset::parseIdentifierToAlias)
+                  .toList();
+          List<Dataset> datasets = tdrService.getDatasetsByIdentifier(identifierList);
+          List<Integer> datasetAliases = datasets
+                  .stream()
+                  .map(Dataset::getAlias)
+                  .toList();
+        // Check that we were able to find a dataset id for all identifiers provided
+        if (identifierList.size() != datasets.size()) {
+            // isolate a list of identifiers that were NOT found
+            List<Integer> notFoundIdentifiers = identifierList
+                    .stream()
+                    .filter(identifier -> !datasetAliases.contains(identifier))
+                    .toList();
+            // throw a NFE to let the client know which identifiers were NOT found so they can rectify their request
+            throw new NotFoundException("Invalid dataset identifiers were provided: " + notFoundIdentifiers);
```

Comment:
When I try this with the following dataset, `DUOS-01234`, I get the following response:
```
{
  "message": "Invalid dataset identifiers were provided: [1234]",
  "code": 404
}
```
It looks like you're using the trimmed version of the alias instead of the values provided by the user for the error message.

Reply:
I found another edge case here. When I used the following: `DUOS-000594, DUOS-000594` (note that they're the same and valid identifiers), I got a confusing error message:
```
{
  "message": "Invalid dataset identifiers were provided: []",
  "code": 404
}
```
I'm split on the right response for this case. On one hand, it's clearly a bad request. However, it's a harmless bad request and one we can easily fix. I would suggest we call distinct the input list, then do the dataset lookups.

Reply:
Updated the not found exception logic to filter down the list of identifier strings instead of integer aliases. I considered using Dataset.parseAliasToIdentifier to rebuild the identifiers from the list of aliases but that would make it possible to return a not found identifier that does not match to the original identifier provided if the one provided does not follow the exact alias format (padded with 6 zeros). That includes the example mentioned in your comment (DUOS-01234).

Also updated the identifier logic to call distinct on the input list as well as the list of alias integers after the strings have been processed. This handles two edge cases: when multiples of the same identifier are provided, and when variations of the identifier string that resolve to the same alias are provided:
examples:
"DUOS-000594, DUOS-000594"
"DUOS-00593, DUOS-000593"
Both of these cases were returning the 404 error with no actual invalid identifiers returned. With the changes both of those inputs would be accepted.